nifi-commons/nifi-sensitive-property-provider/src/main/java/org/apache/nifi/properties/PropertyProtectionScheme.java

```diff
@@ -24,7 +24,8 @@
  * SensitivePropertyProvider.
  */
 public enum PropertyProtectionScheme {
-    AES_GCM("aes/gcm/(128|192|256)", "aes/gcm/%s", "AES Sensitive Property Provider", true);
+    AES_GCM("aes/gcm/(128|192|256)", "aes/gcm/%s", "AES Sensitive Property Provider", true),
+    HASHICORP_VAULT_TRANSIT("vault/transit/[a-z0-9-]+", "vault/transit/%s", "HashiCorp Vault Transit Engine Sensitive Property Provider", false);
```

Comment:
What values should be expected for the parameterized section of the identifier format `vault/transit/%s`?  Different key sizes make sense for AES-GCM, but are there configurable options that are supported?  Following the naming convention, it seems like the format should also start with `hashicorp/vault/transit`, what do you think?

Reply:
This could be any valid path in Vault, and will be the value of vault.transit.path in the bootstrap-hashicorp-vault.conf file.  So if vault.transit.path=nifi-transit, then the scheme will be "hashicorp/vault/transit/nifi-transit", and the SPP will create a Vault path called "transit/keys/nifi-transit".   I chose to let this path be configurable to give the user the flexibility to fit it into an existing Vault naming scheme they might have.  The only mention I can find of character restrictions in Vault key paths is this post, which seems to say only slashes are reserved characters: https://groups.google.com/g/vault-tool/c/_E2N0SLryv8.  However, I chose to restrict this so that it wasn't wide open.  I think this means I need some validation to ensure their configured transit path matches the regex, however, so I'll add that.